### PR TITLE
Hide peer-reviewed papers from participants

### LINKED ISSUE #5666: Add setting to hide peer-reviewed papers
Currently, if contributions are published, papers accepted in peer-review are immediately made available for all participants. We want to add a setting that will keep these papers hidden to others (eg only visible for the submitter and reviewers/managers).

### PR DESCRIPTION
**Pull request by GovernmentPlates**
Closes #5666 

Comment: Not quite sure about the Jinja logic in the `contribution_display.html` - any tips to tidy/neaten this up?

Preview:
Screenshot 2023-02-13 at 17 39 08

image

### CODE REVIEW

File: indico/modules/events/papers/settings.py [hunk 1]
```diff
@@ -54,6 +54,7 @@ def to_python(value):
     'announcement': '',
     'scale_lower': -3,
     'scale_upper': 3,
+    'hide_peer_reviewed_papers': False,
```
Comment (ThiefMaster) [inline indico/modules/events/papers/settings.py:57]: How about just `hide_accepted`? *Anything* in these settings are related to paper peer reviewing ;)

File: indico/modules/events/papers/forms.py [hunk 2]
```diff
@@ -88,6 +88,7 @@ class PaperReviewingSettingsForm(IndicoForm):
     announcement = IndicoMarkdownField(_('Announcement'), editor=True)
     scale_lower = IntegerField(_('Scale (from)'), [InputRequired()])
     scale_upper = IntegerField(_('Scale (to)'), [InputRequired()])
+    hide_peer_reviewed_papers = BooleanField(_('Keep papers hidden'), widget=SwitchWidget(), description=_("Hides papers from participants that have been marked as 'accepted'."))
```
Comment (ThiefMaster) [inline indico/modules/events/papers/forms.py:91]: - Wrap the description into a new line.
- How about "Keep papers hidden from participants even after they have been accepted"? I think that makes it clearer - right now it sounds like the paper would suddenly get hidden once accepted..

File: indico/modules/events/contributions/templates/display/contribution_display.html [hunk 3]
```diff
@@ -273,7 +273,13 @@ <h3>{% trans %}Subcontributions{% endtrans %}</h3>
...
         </section>
     {%- endif %}
-    {% if contribution.event.has_feature('papers') %}
+
+    {% set paper_hidden = contribution.event.cfp.hide_peer_reviewed_papers %}
+    {% set permitted_hidden_paper_viewers = contribution.paper.can_judge(session.user) or contribution.paper.can_review(session.user) or contribution.paper.can_manage(session.user) or contribution.can_submit_proceedings(session.user) %}
+
+    {% if contribution.event.has_feature('papers') and not paper_hidden %}
+        {{ render_paper_section(contribution) }}
+    {% elif paper_hidden and permitted_hidden_paper_viewers %}
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/templates/display/contribution_display.html:282]: Check the `render_paper_section` macro. It already contains some logic, and it should be much easier to extend this to take the new flag into account.

File: indico/modules/events/papers/templates/_contributions.html [hunk 4]
```diff
@@ -3,11 +3,18 @@
 {% macro render_paper_section(contribution) %}
     {% set paper = contribution.paper %}
     {% set can_submit = contribution.can_submit_proceedings(session.user) %}
-    {% set show_paper = paper and paper.accepted_revision %}
```
Comment (ThiefMaster) [inline indico/modules/events/papers/templates/_contributions.html:6]: {% set show_paper = paper and paper.accepted_revision and not paper.event.cfp.hide_accepted %}

Wouldn't that be enough? cc @duartegalvao

Comment (duartegalvao) [reply indico/modules/events/papers/templates/_contributions.html:6]: That would keep the paper hidden for managers/the submitter too, right? They can still access from the timeline I think, but this way would be more convenient

Comment (GovernmentPlates) [reply indico/modules/events/papers/templates/_contributions.html:6]: Yes, it does keep it hidden, but I want to +1 this for easy access.

Comment (ThiefMaster) [reply indico/modules/events/papers/templates/_contributions.html:6]: OK, fine then ;)

File: indico/modules/events/papers/forms.py [hunk 5]
```diff
@@ -88,6 +88,10 @@ class PaperReviewingSettingsForm(IndicoForm):
     announcement = IndicoMarkdownField(_('Announcement'), editor=True)
     scale_lower = IntegerField(_('Scale (from)'), [InputRequired()])
     scale_upper = IntegerField(_('Scale (to)'), [InputRequired()])
+    hide_accepted = BooleanField(_('Keep papers hidden'),
+                                 widget=SwitchWidget(),
+                                 description=_("Keep papers hidden from participants \
+                                               even after they've been accepted."))
```
Comment (ThiefMaster) [inline indico/modules/events/papers/forms.py:94]:
```suggestion
                                 description=_('Keep papers hidden from participants '
                                               "even after they've been accepted."))
```

backslashes are ugly :)

File: CHANGES.rst [hunk 6]
```diff
@@ -35,6 +35,8 @@ Improvements
 - Hide the top box with the latest files of an editable until it has been accepted
   and published (:issue:`5660`, :pr:`5665`)
 - Allow uploading files when requesting changes on the editing timeline (:pr:`5612`)
+- Added a setting to hide peer-reviewed papers from participants even after they have
```
Comment (ThiefMaster) [inline CHANGES.rst:38]:
```suggestion
- Add setting to hide peer-reviewed papers from participants even after they have
```

more concise and we use present in the changelog just like in commit  messages

File: indico/modules/events/papers/templates/reviewing/timeline.html [hunk 7]
```diff
@@ -11,6 +11,11 @@
             {% else %}
                 <div class="paper-files-list flexrow f-a-baseline">
                     <div class="paper-files-list-label f-self-no-shrink">
+                        {% if paper.event.cfp.hide_accepted %}
+                            <span title="{% trans %}Accepted papers are currently hidden from participants{% endtrans %}"
+                                class="category-action left icon-eye-blocked active" style="padding-right:0.2rem;">
```
Comment (ThiefMaster) [inline indico/modules/events/papers/templates/reviewing/timeline.html:16]:
```suggestion
                            <span title="{% trans %}Papers are currently hidden from other participants{% endtrans %}"
                                  class="category-action left icon-eye-blocked active"
                                  style="padding-right:0.2rem;">
```

Similar to the previous improvements, it's clearer that way.
Also fixed the indentation/wrapping.

File: indico/modules/events/papers/templates/reviewing/timeline.html [hunk 8]
```diff
@@ -11,6 +11,12 @@
             {% else %}
                 <div class="paper-files-list flexrow f-a-baseline">
                     <div class="paper-files-list-label f-self-no-shrink">
+                        {% if paper.event.cfp.hide_accepted %}
+                            <span title="{% trans %}Papers are currently hidden from other participants{% endtrans %}"
+                                  class="category-action left icon-eye-blocked active"
+                                  style="padding-right:0.2rem;">
```
Comment (ThiefMaster) [inline indico/modules/events/papers/templates/reviewing/timeline.html:17]:
```suggestion
                                  style="padding-right: 0.2rem;">
```

**Pull request merged**